### PR TITLE
[SiWx917] Adding the sleepy support for 917 SoC

### PR DESCRIPTION
**Pull request by chirag-silabs**
#### Problem / Feature
What is being fixed? What is the feature being added? Examples:
Enabling Sleepy on 917 SoC

#### Change overview
 - Adding the hardware setup call, which powers down the peripherals once sleep is called.
 - Borrowing the sleep file for the WiFi SDK which should be fixed in the next release of the wifi sdk.
1. Delay present after sleep which is not needed.
https://github.com/SiliconLabs/wiseconnect/blob/master/components/device/silabs/si91x/mcu/drivers/peripheral_drivers/src/sl_si91x_m4_ps.c#L344 

2. Conflicting of IRQ021 handler since it is already added when btn drivers are added 
https://github.com/SiliconLabs/wiseconnect/blob/master/components/device/silabs/si91x/mcu/drivers/peripheral_drivers/src/sl_si91x_m4_ps.c#L113-L118

#### Testing
Tested manually

### CODE REVIEW

**Review by selissia — APPROVED** [review]

File: matter/si91x/siwx917/BRD4338A/autogen/sl_event_handler.c [hunk 1]
```diff
@@ -18,6 +19,9 @@ void sl_platform_init(void) {
   sli_si91x_platform_init();
   RSI_Board_Init();
   DEBUGINIT();
+#if SL_ICD_ENABLED && !(defined(DISPLAY_ENABLED))
```
Comment (jmartinez-silabs) [inline matter/si91x/siwx917/BRD4338A/autogen/sl_event_handler.c:22]: Did you test that SL_ICD_ENABLED exists at this level? I am not 100% sure it does

File: matter/si91x/siwx917/BRD4338A/support/inc/sl_si91x_m4_ps.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/***************************************************************************/ /**
```
Comment (jmartinez-silabs) [inline matter/si91x/siwx917/BRD4338A/support/inc/sl_si91x_m4_ps.h:1]: Can you fix the comment header. Remove the / / at the end of this line and styler should place everything back. Same for the .c file.

File: matter/si91x/siwx917/BRD4338A/autogen/sl_event_handler.c (hunk 1 above)
Comment (chirag-silabs) [reply matter/si91x/siwx917/BRD4338A/autogen/sl_event_handler.c:22]: I have tested and it is working on this level.

File: matter/si91x/siwx917/BRD4338A/support/inc/sl_si91x_m4_ps.h (hunk 2 above)
Comment (chirag-silabs) [reply matter/si91x/siwx917/BRD4338A/support/inc/sl_si91x_m4_ps.h]: Done

**Pull request merged**